src/step.rs

```diff
@@ -96,45 +97,48 @@ where
     F: Fn(&InterpCx) -> ShouldContinue,
 {
     let mut message = None;
-    loop {
-        if pcx.ecx.stack().len() <= 1 && is_ret(&pcx.ecx) {
-            break;
-        }
-        match pcx.ecx.step() {
-            Ok(true) => {
-                *pcx.step_count += 1;
-                crate::watch::step_callback(pcx);
-
-                if let Some(frame) = pcx.ecx.stack().last() {
-                    if let Some(block) = frame.block {
-                        let blck = &frame.body.basic_blocks()[block];
-                        if frame.stmt != blck.statements.len()
-                            && crate::should_hide_stmt(&blck.statements[frame.stmt])
-                            && !pcx.config.bptree.is_at_breakpoint(&pcx.ecx)
-                        {
-                            continue;
-                        }
-                    } else {
-                        // Unwinding, but no cleanup for this frame
-                        // FIXME make step behaviour configurable
+    let ret: InterpResult<'_, _> = (|| {
+        loop {
+            if pcx.ecx.stack().len() <= 1 && is_ret(&pcx.ecx) {
```

Comment:
This would now prevent any thread from exiting. Please only do this when there is only one thread left.